nix/nim-status/build.nix

```diff
@@ -0,0 +1,317 @@
+    else throw "Unsupported platform!";
+
+  sqliteCompilerFlags = if isAndroid then
+  "-isysroot ${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/${osId}-${osArch}/sysroot -target ${androidTarget}${api}"
+  else if isIOS then
+  "-isysroot $(xcrun --sdk ${iosSdk} --show-sdk-path) -miphonesimulator-version-min=7.0 -fembed-bitcode -arch ${iosArch}"
+  else throw "Unsupported platform!";
+
+  sqliteLinkerFlags = if isAndroid then
+  " --sysroot ${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/${osId}-${osArch}/sysroot -target ${androidTarget}${api} -v -Wl,-soname,libstatus.so"
+  else if isIOS then
+  "--sysroot $(xcrun --sdk ${iosSdk} --show-sdk-path) -miphonesimulator-version-min=7.0 -fembed-bitcode -arch ${iosArch}"
+  else throw "Unsupported platform!";
+
+  hostMap = {
+      "386" = "x86";
+      "arm" = "arm";
+      "arm64" = "aarch64";
+    };
+  hostFlag = if isAndroid then androidTarget else lib.getAttr arch hostMap;
+
+
+  ldDirMap1 = {
+    "386" = "i686-linux-android";
+    "arm" = "arm-linux-androideabi";
+    "arm64" = "aarch64-linux-android";
+  };
+
+  linkerFlags = if isAndroid then
+  "--sysroot ${ANDROID_NDK_HOME}/platforms/android-${api}/arch-${ldArch} -target ${androidTarget}"
+  else if isIOS then
+  "--sysroot $(xcrun --sdk ${iosSdk} --show-sdk-path) -fembed-bitcode -arch ${iosArch}"
+  else throw "Unsupported platform!";
+
+  cppLibPathPrefix = "${ANDROID_NDK_HOME}/sources/cxx-stl/llvm-libc++/libs/${lib.getAttr arch ldDirMap}/";
+
+  cppLibPathPrefix1 = "${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/${osId}-${osArch}/sysroot/usr/lib/${lib.getAttr arch ldDirMap1}/${api}/";
+
+  androidToolPathPrefix = "${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/${osId}-${osArch}/bin";
+  androidToolPrefix = "${androidToolPathPrefix}/${targetArch}-linux-${platform}";
+  compilerVars = if isAndroid then
+    ''
+      export PATH=${androidToolPathPrefix}:$PATH
+      export AR=${androidToolPrefix}-ar
+      export AS=${androidToolPrefix}-as
+      export NM=${androidToolPrefix}-nm
+      export RANLIB=${androidToolPrefix}-ranlib
+      export CC=${androidToolPathPrefix}/clang
+
+      # This is important, otherwise Nim might not use proper tooling
+      mkdir bin
+      ln -s $AR bin/ar
+      ln -s $AS bin/as
+      ln -s $CC bin/gcc
+      ln -s $RANLIB bin/ranlib
+      export PATH=./bin:$PATH
+    ''
+    else if isIOS then
+    ''
+      export PATH=${xcodeWrapper}/bin:$PATH
+      export CC=$(xcrun --sdk ${iosSdk} --find clang)
+      export CXX=$(xcrun --sdk ${iosSdk} --find clang++)
+      export CFLAGS="${compilerFlags}"
+      export LDFLAGS="${linkerFlags}"
+    ''
+    else throw "Unsupported platform!";
+ 
+  # src = pkgs.fetchgit {
+  #   url = "https://github.com/status-im/nim-status";
+  #   rev = "33241ee463291ed912681eff278658dc1dba1dcb";
+  #   sha256 = "1ki32s13zrwjzhbdyzfzj29mj3hkib2iwqwcwvlan9699p4v19iq";
+  #   leaveDotGit = true;
+  #   fetchSubmodules = false;
+  # };
+
+  opensslSrc = stdenv.mkDerivation rec {
```

Comment:
I don't see you using this, or `pcreSrc`, or the commented stuff above.
Maybe run [`nix-linter`](https://github.com/Synthetica9/nix-linter) on the files you've changed:
```
 > nix-env -iA nixos.nix-linter       
warning: name collision in input Nix expressions, skipping '/home/sochan/.nix-defexpr/channels_root/nixos'
warning: name collision in input Nix expressions, skipping '/home/sochan/.nix-defexpr/channels_root/nixos-unstable'
installing 'nix-linter-unstable-2020-09-25'
building '/nix/store/2bjxnhviha8ysrrvm0xx2bw3xgxr2apd-user-environment.drv'...
created 2 symlinks in user environment

 > nix-linter nix/nim-status/build.nix
Unused `let` bind `androidIncludes` at nix/nim-status/build.nix:79:3
Unused `let` bind `opensslSrc` at nix/nim-status/build.nix:186:3
Unused `let` bind `pcreSrc` at nix/nim-status/build.nix:202:3
Unused argument `fetchFromGitHub` at nix/nim-status/build.nix:1:1-318:1
Unneeded `rec` on set at nix/nim-status/build.nix:24:19-28:4
LetInInheritRecset at nix/nim-status/build.nix:15:1-318:1
```